### DISCUSSION TITLE
Ring Not Working In Production

### DISCUSSION THREAD

**Discussion by raymolla7** [tailwindlabs/tailwindcss#8492] [Help]
Can't get Tailwind Ring working on production no matter what I try. (This has been going on for a month 😔)

Tailwind Version: 3.0.24
React: 17.0.2
NextJS: 12.0.8
PostCSS: 8.4.5

Here is an example on Prod I tried adding but it is not showing up: 

https://hideandseek.world/faq

If you inspect is shows the classes are there:
![image](https://user-images.githubusercontent.com/56936621/171554230-9c30e650-51d0-43c1-b325-39304981e30f.png)

Should show up with the blue ring around like this (it works in Local):
<img width="94" alt="image" src="https://user-images.githubusercontent.com/56936621/171553666-ffa45744-9501-4458-ad41-c370bac9fdca.png">

I've seen others with similar issues: https://github.com/tailwindlabs/tailwindcss/issues/6328

But tried everything and still not working.

EDIT: I eventually ended up using tailwind Outline instead of Ring. Outline seems to work in production.

**Comment by thecrypticace** [comment]
Hey, we've been somewhat following that issue but as far as Tailwind CSS itself goes it's producing the correct CSS. I suspect this is all down to outdated tooling / CSS minifiers. Would you mind providing either:
- Details about what tools you're using to build the site (exact versions please)
- A repository that demonstrates the issue

I'm aware that it only reproduces in production so if it doesn't reproduce in the dev server that is entirely okay. We can handle using the tool(s) to build for production and testing it that way.

**Comment by PositiveUser** [comment]
You are using `img` element and you are maybe using `box-shadow` with `inset`, which does not work for images.

[Check this Tailwind Play](https://play.tailwindcss.com/eHD8qmNaBH)

**Reply by raymolla7** [reply]
No this doesn't work in Production regardless if it's an `img` or `div`.  It works in local so not sure what the issue is. I think thecrypticace might be correct this is probably something to do with tooling / CSS minifiers.

**Comment by guillaumeprevost** [comment]
Not really an answer, but have the same issue and maybe hints to help finding the issue.

I can see that in production the classes are there, and they seem to be well defined

Here are the screenshots : 

Local :
![image](https://user-images.githubusercontent.com/4436023/179760825-40af0e64-a5a9-4365-a2c4-6c252c9703ea.png)

Production :
![image](https://user-images.githubusercontent.com/4436023/179760981-830f5af5-a164-4160-b833-323d83495285.png)

![image](https://user-images.githubusercontent.com/4436023/179761238-b6a5042f-93c0-40b5-8365-bc2aad798a62.png)

I use this in my project : 
- VueJS 2 + [NuxtJS 2.15.8](https://www.npmjs.com/package/nuxt) which uses webpack for packaging
- TailwindCSS with [vue-tailwind 2.5.1](https://www.npmjs.com/package/vue-tailwind)

In production only, the following webpack options are enabled : extractCSS, optimizeCSS, html.minify.minifyCSS.

**Reply by raymolla7** [reply]
I couldn't figure out the issue, eventually I just ended up using tailwind Outline instead of Ring

**Reply by xithalius** [reply]
We had the same problem on our production build. I stumbled upon [this comment from Adam Wathan](https://github.com/tailwindlabs/tailwindcss/issues/6328#issuecomment-1146588541). Disabling our optimization block in our webpack config did indeed solve the issue because of the outdated cssnano. We decided to upgrade from CRA 4 to 5 which will use cssnano 5 because of the upgrade to Webpack 5.

**Reply by guillaumeprevost** [reply]
Thanks to both of you for your replies

@raymolla7 unfortunately "ring" and "outline" don't behave exactly in the same way in CSS. 
In our case, not using ring offset the rest of the content around the element.

@xithalius upgrading to Webpack 5 doesn't seem to be an option when using Nuxt (2.x) https://github.com/nuxt/nuxt.js/issues/8252

**Comment by TeddyBurnsides** [comment]
Also ran into this issue. Basically exactly the same scenarios as described above but using Tailwind with the create-react-app bootstrap. Ultimately we switched to outline and didn't see any issues.

**Comment by devenbhooshan** [comment]
if anybody is using `create-react-app` and is facing this issue, please upgrade your `react-scripts` to `5.X.y`. It worked for me.
